### PR TITLE
docs: fixed mistakes, removed [&]

### PR DESCRIPTION
**Pull request by Henonicks**
I see some examples use `[&]` and the advanced reference says that it's better not to use it. So I don't think we should do it in the examples. Also, there were punctuation mistakes in those files, fixed those.
- [X] My pull request is made against the `dev` branch.
- [X] I have ensured that the changed library can be built on your target system. I did not introduce any platform-specific code.
- [X] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [X] I tested my commits, by adding a test case to the unit tests if needed
- [X] I have ensured that I did not break any existing API calls.
- [X] My code follows the [coding style guide](https://dpp.dev/coding-standards.html) (if you are not sure, match the code style of existing files including indent style etc).
- [X] I have not built my pull request using AI, a static analysis tool or similar without any human oversight. Where I have generated this pull request using a tool, I have justified why this is needed.

### CODE REVIEW

File: docpages/advanced_reference/lambdas_and_locals.md [hunk 1]
```diff
@@ -2,11 +2,11 @@
...
+It is important to remember that when you put a lambda callback onto a function in D++, that this lambda will execute at some point in the **future**. As with all things in the future and as 80s Sci-Fi movies will tell you, when you reach the future, things may well have changed!
 
 \image html delorean-time-travel.gif
 
-To explain this situation and how it causes issues I'd like you to imagine the age old magic trick, where a magician sets a fine table full of cutlery, pots, pans and wine. He indicates to the audience that this is authentic, then with a whip of his wrist, he whips the tablecloth away, leaving the cutlery and other tableware in place (if he is any good as a magician!)
+To explain this situation and how it causes issues, I'd like you to imagine the age-old magic trick, where a magician sets a fine table full of cutlery, pots, pans and wine. He indicates to the audience that this is authentic, then with a whip of his wrist, he whips the tablecloth away, leaving the cutlery and other tableware in place (if he is any good as a magician!)
```
Comment (Jaskowicz1) [inline docpages/advanced_reference/lambdas_and_locals.md:9]: "age-old" is correct.

**Review by Jaskowicz1 — COMMENTED** [review]
Just one thing :)

**Comment by Jaskowicz1** [comment]
I would advise that you move your future PRs into a separate branch. Your PRs are starting to contain many commits from old PRs (because you're merging them into your dev branch).

**Comment by Henonicks** [comment]
But if I do so, some files will be missing and I'll request pull without needed/non-up-to-date files, no?

File: docpages/advanced_reference/lambdas_and_locals.md (hunk 1 above)
Comment (Henonicks) [reply docpages/advanced_reference/lambdas_and_locals.md:9]: I Googled it and it says "age-old" all over, there is no "age old".

**Comment by ruslan-ilesik** [comment]
you can create new branch which will be copy of current dev branch and use `git cherry-pick` (more detailed [here](https://stackoverflow.com/questions/9339429/what-does-cherry-picking-a-commit-with-git-mean))

**Comment by Jaskowicz1** [comment]
> But if I do so, some files will be missing and I'll request pull without needed/non-up-to-date files, no?

No. If you make your branch from dev, you'll be okay. Even if you don't, if you only commit your files, only they get seen by GitHub.

**Comment by Jaskowicz1** [comment]
> you can create new branch which will be copy of current dev branch and use `git cherry-pick` (more detailed [here](https://stackoverflow.com/questions/9339429/what-does-cherry-picking-a-commit-with-git-mean))

You don't need `git cherry-pick`. Just do `git checkout -b branch-name dev`

File: docpages/advanced_reference/lambdas_and_locals.md (hunk 1 above)
Comment (Jaskowicz1) [reply docpages/advanced_reference/lambdas_and_locals.md:9]: Oh I read this wrong nvm. I thought you changed it from `age-old` to `age old`. You can resolve this comment.

**Comment by Henonicks** [comment]
> > But if I do so, some files will be missing and I'll request pull without needed/non-up-to-date files, no?
> 
> No. If you make your branch from dev, you'll be okay. Even if you don't, if you only commit your files, only they get seen by GitHub.

So if I copy it to my own branch and commit files that I need to only, I'll be fine and I won't have to keep pulling/re-cloning?

**Comment by Jaskowicz1** [comment]
> So if I copy it to my own branch and commit files that I need to only, I'll be fine and I won't have to keep pulling/re-cloning?

Exactly. This means you just do what you want on your branch and once the PR is merged, you delete the branch. You should still keep pulling dev though, to make sure that's up-to-date.

**Comment by Henonicks** [comment]
> Exactly. This means you just do what you want on your branch and once the PR is merged, you delete the branch. You should still keep pulling dev though, to make sure that's up-to-date.

So, I "copy" the branch, do stuff and when it's merged, delete it and do it over again?

**Comment by Jaskowicz1** [comment]
> > Exactly. This means you just do what you want on your branch and once the PR is merged, you delete the branch. You should still keep pulling dev though, to make sure that's up-to-date.
> 
> So, I "copy" the branch, do stuff and when it's merged, delete it and do it over again?

Yup. If you feel like you need more information about this, I can look to make a tutorial for this.

**Comment by Henonicks** [comment]
> Yup. If you feel like you need more information about this, I can look to make a tutorial for this.

I guess I'll just keep on learning the power of Googling.

File: docpages/advanced_reference/lambdas_and_locals.md (hunk 1 above)
Comment (Henonicks) [reply docpages/advanced_reference/lambdas_and_locals.md:9]: Holy moly, first try pr?

**Review by braindigitalis — APPROVED** [review]

**Review by Mishura4 — APPROVED** [review]

**Pull request merged**